ideagenerator/generator.js

```diff
@@ -5,18 +5,27 @@ let recently_used = [];
 let character_is_group = false;
 let character_post_description = '';
 
-window.onload = function onLoad() {
+function parseCategories() {
 	// Store list of entries by category name
 	for (let i = 0; i < category_names.length; i ++) {
 		name = category_names[i];
 		categories[name] = getCategory(name);
 	}
 }
 
+let parsed = false;
+
 function generate() {
-	reset();
-	let template = pickRandom('template');
-	let result = fillInTemplate(template);
+	if(!parsed)
+		parseCategories();
+	let result;
+	do
```

Comment:
```suggestion
	do {
```

Reply:
I personally prefer K&R so I'm mostly used to that.